Log Errors when ValidateMigrations cannot parse the response

paytonrules opened the pull request:
Sometimes Github is erroring on unmarshal, so turn it into a straight string for debugging.

In addition, only run validatemigrations on a PR. This will help with occasional API throttling we run into with github.

paytonrules committed:
Try to log the bytes

Sometimes Github is erroring on unmarshal, so turn it into a straight string.

rmulhol commented on cmd/checkMigrations.go:
```diff
@@ -114,7 +114,7 @@ func getGithubFileNames() ([]string, error) {
 
 	if jsonErr != nil {
-		return []string{}, fmt.Errorf(errorContext, jsonErr)
+		return []string{}, fmt.Errorf("Failed to unmarshal %s, %w", string(body), jsonErr)
```
super minor but I think technically we want to avoid capitalization in error messages: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

paytonrules replied:
Bah humbug. Fixed.

paytonrules committed:
Don't capitalize errors

Golang standards

paytonrules committed:
Only run validatemigrationorder on PR builds

Also move it to earlier in the build since the rest is unneeded if the
script fails.
